### ISSUE TITLE
Linking notes from same searches or same paper

### ISSUE THREAD

**Issue by ajburls** [dwyl/best-evidence#162]
As a user of the app, I want to be able to write notes on a search and see those notes and add to them if I run the same search again. While at the moment it is easy to find a previous search and open the notes attached to it, if a user has a longer history and does more searches, this will not be so easy, so we need to link the searches in the background. 

Similarly, if I have found a paper and, say, critically appraised it and recorded this in the notes, I need to see these notes when the same paper is identified in a subsequent search (whether a re-run of the same search or a separate search), so that I don't have two sets of notes about the same paper.

**Comment by SimonLab** [comment]
Concerning your second point @ajburls ,if I'm correct, at the moment the notes are unique per paper/publication. So if a user add a note on a specific paper this note will be the same on the other searches. For example on the screenshot below the note on "Say Hello To Heart Nuts" papers is the same for the two searches "hello":
![paper_note](https://user-images.githubusercontent.com/6057298/29572718-0315fee6-8755-11e7-9813-03577e185145.png)
@ajburls could you confirm this is working for you on staging?

Concerning your first point I think we need a bit more thought on how we define "same search". We spoke about this during our last demo but it might be good to write down our ideas on this issues. For papers/publication we can identify easily if they are unique as Tripdatabase API gives us a unique id. However for searches are a bit more complicated as they are free text. One easy way to determine if a search already exists in the database is to simply compare the search terms together. However if the user insert a typo in the search term the database will consider this as a new search. Also do you consider a search terms with different cases to be the same, for example should "hello", "HELLO", "HeLlO"... be the same term search? This point is also linked to the following comment: https://github.com/dwyl/best-evidence/issues/22#issuecomment-323777869

As this feature needs a bit more thought and refactoring (new tables in the database and build the logic of the code) I'm not sure it could be done in the sprint 3 at the moment. But let us know if this is something you would like to see implemented soon and we can see if we could make place in the sprint by removing other less high priority issues.

Let me know your thoughts or if my comment is unclear, thanks

**Comment by markwilliamfirth** [comment]
@ajburls Simon's currently working on 165 - I didn't realise 162 wasn't in sprint 3 when it was mentioned earlier - can you confirm if this is something you'd want in sprint 3 and the priority for it? I can see you've put it 7th in the list in the sprint planning milestone - is that correct?

**Comment by ajburls** [comment]
@markwilliamfirth I think the basic issue of Linking notes from same searches or same paper has been done as my test above confirmed. @SimonLab can you confirm that there is nothing more needed to be done on this and close issue if so?

I will open a separate new issue for the more complex dating of searches which is not a high priority for the moment.

**Comment by SimonLab** [comment]
@ajburls having a unique note for the same publications is done, however I still need to implement unique for same searches, this depend on #165 as I need to link the information of the user to the search when a user scroll the page.

**Comment by ajburls** [comment]
@SimonLab ok

**Comment by SimonLab** [comment]
@ajburls is this issue still a priority 3? We talk in our standup that you would like the same note to be liked for the same searches done on the sprint 3. I will help us to know how important this is so we can focus on the right issues, could you update the priority label if necessary, thanks

**Comment by ajburls** [comment]
I think that having one set of notes per unique search, even if that search is repeated is a priority 3, if that is what you are asking for clarification on @SimonLab . This doesn't require resolution of the date issue does it? Or were you asking about something else?

**Comment by SimonLab** [comment]
Thanks for the priority confirmation. 
To clarify what I understand for the "same search note" feature:
We can detect if a search is the same just by comparing the term (case insensitive). If the search is the same we will then link the note. On the history page we could at this stage only show one time the search term (instead of multiple same ones) and we can order the searche terms by the last date the term search has been done. This order might resolve the date issue?
I hope my though make sense?

**Comment by ajburls** [comment]
@SimonLab this makes perfect sense. I think this will be an adequate compromise for the MVP version of the app.

**Comment by markwilliamfirth** [comment]
>we can order the search terms by the last date the term search has been done. This order might resolve the date issue?

@SimonLab do we need to create a new issue for this one?

**Comment by SimonLab** [comment]
@ajburls This is now ready to be tested (link same searches). Unfortunately I had to reset the database to get the last structure for the data so you will have to create a new account again.

**Comment by ajburls** [comment]
I note history reset.that answers my previous comment. Can't test this because trying to run some search again to have two searches for testing returned internal server error



Sent from my Samsung Galaxy smartphone.
-------- Original message --------

**Comment by SimonLab** [comment]
Thanks for testing this. Could you try again, you might need to logout and to be sure you create a new account or login with a new one as I reseted the database. Let me know if you still have the server error, thanks

**Comment by ajburls** [comment]
@SimonLab Managed to get my computer working. Tested on computer and there is a single note for each paper. Latest search goes to the top as we agreed. Great :-)

I note that notes on papers, because they are linked to the search, also move up the order list (currently only saying today) , presumably if I run the same search tomorrow, notes on a paper made today will appear as if there were made tomorrow). Is that right? Or will they keep the last date they were edited

The latter is better.

**Comment by SimonLab** [comment]
Just to recap:
My understanding is that duplicate searches should be avoid on the history page, see the following comments:
- https://github.com/dwyl/best-evidence/issues/162#issuecomment-326561465
- https://github.com/dwyl/best-evidence/issues/22#issuecomment-323777869

So at the moment the search terms are ordered by the most recent term searched. The publications for a search are still linked to the term so they follow the order of the search list.

At the moment we are saving the two different times for each search:
- the date when the search was first done
- the date of the last time a search as been done

I think this issue needs more details and maybe some feedback from some users. We also need to understand what problem we try to resolve for the users with the history page. As you said on our standup @ajburls I think you have the idea of creating a report from the history and notes. The history page could definitely help building this report but maybe we should try to avoid merging the two features (history and report) together as it might be confusing.
It might be good to have a clear idea what we want to achieve and maybe we should start with some wireframes and a detailed user flow.

**Comment by markwilliamfirth** [comment]
This functionality has been completed; this issue is now closed in favour of:

https://github.com/dwyl/best-evidence/issues/204
https://github.com/dwyl/best-evidence/issues/205
